pkg/plugins/manager/installer.go

```diff
@@ -54,89 +54,90 @@ func (m *PluginInstaller) Add(ctx context.Context, pluginID, version string, opt
 		return err
 	}
 
+	archive, err := m.install(ctx, pluginID, version, compatOpts)
+	if err != nil {
+		return err
+	}
+
+	for _, dep := range archive.Dependencies {
+		m.log.Info(fmt.Sprintf("Fetching %s dependency %s...", pluginID, dep.ID))
+
+		err = m.Add(ctx, dep.ID, dep.Version, opts)
+		if err != nil {
+			return fmt.Errorf("%v: %w", fmt.Sprintf("failed to download plugin %s from repository", dep.ID), err)
+		}
+	}
+
+	_, err = m.pluginLoader.Load(ctx, sources.NewLocalSource(plugins.ClassExternal, []string{archive.Path}))
```

Comment:
curious: what's this check for?

Reply:
This is the code that actually loads the new plugin(s) once they've been extracted to disk